src/state/migrations.ts

```diff
@@ -5,12 +5,15 @@ import { migration0 } from './migrations/0';
 import { migration1 } from './migrations/1';
 import { migration2 } from './migrations/2';
 import { migration3 } from './migrations/3';
+import { migration4 } from './migrations/4';
 
 export const migrations: MigrationManifest = {
   0: migration0,
   1: migration1,
   2: migration2,
   3: migration3,
+  // @ts-expect-error PersistAppStateV4 is a union of PersistedState and new app state
```

Comment:
why is the compiler angry?

Reply:
The fn expects PersistedState and not a union of `PersistedState & { appUiConfigs: ... }`

Reply:
the other ones did the same thing, why is it failing now?

Reply:
I think I'd prefer the function be the type this thing wants (take PersistedState | undefined as the argument) then inside the function cast PersistedState to PersistAppStateV3. That's generally safer than totally turning off the compiler checks for a line.

Reply:
here's someone else doing a similar thing:
```
const manifest: MigrationManifest = {
    1: (state: PersistedState) => {
        const persistedState = state as Root<VERSION_0>;
        const migratedState: Root<VERSION_1> = migrateState1(persistedState);
        return migratedState;
    },
    2: (state: PersistedState) => {
        const persistedState = state as Root<VERSION_1>;
        const migratedState: Root<VERSION_2> = migrateState2(persistedState);
        return migratedState;
    },
    3: (state: PersistedState) => {
        const persistedState = state as Root<VERSION_2>;
        const migratedState: Root<VERSION_3> = migrateState3(persistedState);
        return migratedState;
    }
}
```

Reply:
Helper function added to cast param types